packages/supabase_flutter/test/supabase_flutter_test.dart

```diff
@@ -93,11 +94,32 @@ void main() {
 
   group('Deep Link with PKCE code', () {
     late final PkceHttpClient pkceHttpClient;
+    late final bool mockEventChannel;
+
+    /// Check if the current version of AppLinks uses an explicit call to get
+    /// the initial link. This is only the case before version 6.0.0, where we
+    /// can find the getInitialAppLink function.
+    bool appLinksExposesInitialLinkInStream() {
+      try {
+        // before app_links 6.0.0
+        (AppLinks() as dynamic).getInitialAppLink;
+        return false;
+      } on NoSuchMethodError catch (_) {
+        return true;
+      }
+    }
```

Comment:
Instead of checking the current environment to see if app_links 6.0.0 is used or not, could we create two test cases, one where the behavior of app_links v6.0.0 is used, and one where app_link version prior to that is used?

Reply:
The GH Actions are already set up in a way that the tests are run both for the new app_links version and for the old one.

What could be possible is to split up the test cases and have one of them be skipped based on the environment (app_links version). We could then still use this same logic to decide which one is to be skipped and which one to be run. That might make a little more sense than having one test that does different things based on the environment. 

Would that solve it for you?

Reply:
Could you elaborate on how GitHub actions are already set up in a way that tests are for both of the new and old app_links versions?

Reply:
Ah, alright! Could we maybe keep a comment mentioning that the CI is setup to test new and old versions of app_links then so that it's obvious?